### PR TITLE
release: 2.10.0

### PR DESCRIPTION
**Pull request by Saviq**
- ABI summary:
  - miral ABI unchanged at 5
  - mircommon ABI unchanged at 9
  - mircookie ABI unchanged at 2
  - mircore ABI unchanged at 2
  - miroil ABI unchanged at 2
  - mirplatform ABI unchanged at 23
  - mirserver ABI unchanged at 58
  - mirwayland ABI unchanged at 3
  - mirplatformgraphics ABI unchanged at 20
  - mirinputplatform ABI unchanged at 8
- Enhancements:
  - Verify wl_pointer.set_cursor() serial matches latest
    wl_pointer.enter() serial (#2532)
  - MinimalWindowManager: allow moving window from the maximized
    state (#2593)
  - X11 platform: send discrete scroll (#2579)
  - Verify serials for move/resize requests (#2622)
  - Modernize touch events (#2623)
  - Implement Modifier+Drag window move gesture (#2626)
  - Improve Client class lifetime management (#2644)
  - Implement X11 platform pixel format selection correctly (#2648)
  - Deactivate text input in destructor for v1 and v2 protocols (#2657)
  - gbm-kms/quirks: Quirk off AST devices (#2679)
- Bugs fixed:
  - Incorrect rendering when a surface spans multiple outputs (#1753)
  - [Xwayland] weird focus problems with CLion (#2255)
  - Maximized windows end up behind panels after screen going off (#2580)
  - Menus of panels don't close when losing (or don't lose) focus (#2585)
  - CapsLock state gets stuck (#2592)
  - Popups dismissed by Mir when parent clicked (#2604)
  - Latest swaybg (on 22.10) doesn't paint on Mir (#2620)
  - Shutdown crash when OSK is running (#2639) 
  - Cursor only updates once per enter on XWayland (#2643)
  - OSK stays on screen with unfocused app (#2651)
  - Scaling messes up screencopy (#2653)
  - Mir sometimes shuts down when moving to a different TTY when
    logind-driven (#2661)
  - Bug in implementation of xdg_output (#2666)
  - Waking screens up after idle-timeout often stops
    or locks Mir up (#2674)
  - In CLion cut & paste only works up to the last \n in the source
    text (#2677)

### CODE REVIEW

**Comment by Saviq** [comment]
Platform | 22.04LTS | 21.10 | 20.04LTS
-- | -- | -- | --
gbm-kms | @Saviq |   |  
gbm-x11 | @Saviq |   |  
eglstream-kms |   |   |  
dispmanx |   |   | 

Console | 22.04LTS | 21.10 | 20.04LTS
-- | -- | -- | --
minimal |   |   |  
linux-VT |   |   |  
logind | @Saviq |   |  

WM | 22.04LTS | 21.10 | 20.04LTS
-- | -- | -- | --
floating | @Saviq |   |  
tiling |   |   |  
kiosk |   |   |

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,53 @@
...
+
+  * New upstream release 2.10.0
+
+    - ABI summary:
+      . miral ABI unchanged at 5
+      . mircommon ABI unchanged at 9
+      . mircookie ABI unchanged at 2
+      . mircore ABI unchanged at 2
+      . miroil ABI unchanged at 2
+      . mirplatform ABI unchanged at 23
```
Comment (AlanGriffiths) [inline debian/changelog:11]: I don't think the mirplatform changes (`EventBuilder` gained a virtual function; `WriteMappableBuffer` gained a virtual base class; and, `BufferRenderTarget::set_buffer()` changed signature) are ABI safe.

**Review by AlanGriffiths — APPROVED** [review]
I think we need to bump mirplatform

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+      . mirserver ABI unchanged at 58
+      . mirwayland ABI unchanged at 3
```
Comment (AlanGriffiths) [inline debian/changelog:13]: We also have changes to the mirwayland API

**Comment by Saviq** [comment]
I believe we should look into this before release:
- #2698

**Comment by Saviq** [comment]
~rc2

[TestPlan](https://wiki.ubuntu.com/Process/Merges/TestPlans/Mir)

Platform | 22.04LTS | 21.10 | 20.04LTS
-- | -- | -- | --
gbm-kms | @Saviq |   |  
gbm-x11 | @Saviq |   |  
eglstream-kms |   |   |  
dispmanx |   |   | 

Console | 22.04LTS | 21.10 | 20.04LTS
-- | -- | -- | --
minimal |   |   |  
linux-VT |   |   |  
logind | @Saviq |   |  

WM | 22.04LTS | 21.10 | 20.04LTS
-- | -- | -- | --
floating | @Saviq |   |  
tiling |   |   |  
kiosk |   |   |

**Comment by AlanGriffiths** [comment]
https://github.com/MirServer/mir/actions/runs/3241579301/jobs/5313712634

Seems we may also need #2696?

**Review by AlanGriffiths — APPROVED** [review]
bors delegate

**Comment by AlanGriffiths** [comment]
Rebooting release

**Pull request closed without merging**